IQSS/dataverse
# 9228 - add OIDC development setup for OIDC login feature testing

**What this PR does / why we need it**:

As stated in #9228, developers currently don't have a way to test the [login with OIDC](https://guides.dataverse.org/en/5.12.1/installation/oidc.html) feature of Dataverse.

This PR includes a dockerized Keycloak setup with OIDC support for development purposes, as well as its associated documentation.

TODOs.

- [X] Add Keycloak-Docker config
- [x] Add Keycloak-Docker documentation at (future version of) https://guides.dataverse.org/en/5.12.1/developers/remote-users.html
- [x] Test current OIDC login flow with this setup

**Which issue(s) this PR closes**: 

Closes #9228

**Special notes for your reviewer**:
Not yet.

**Suggestions on how to test this**:

Follow the next steps:

1. (Positioned in `/conf/keycloak/`) Run Keycloak docker container: You can use _docker-compose_ file or _run-keycloak.sh_ script. 
2. Verify that Keycloak is running on http://localhost:8090/.
3. Verify that Dataverse is locally running
4. (Positioned in `/conf/keycloak/`) Execute the following API call which enables the Keycloak OIDC client as an authentication provider for Dataverse:  `curl -X POST -H 'Content-type: application/json' --upload-file oidc-keycloak-auth-provider.json http://localhost:8080/api/admin/authenticationProviders`
5. Test that the new OIDC auth flow works. A new option "OIDC-Keycloak" should appear on the login screen. The credentials for the Keycloak test user are: kcuser / kcpassword (usr/pwd).

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No.

**Is there a release notes update needed for this change?**:
Not sure.

**Additional documentation**:
Not yet.
